src/renderer/widgets/Image.cpp

```diff
@@ -1,7 +1,64 @@
 #include "Image.hpp"
 #include "../Renderer.hpp"
+#include "../../core/hyprlock.hpp"
 #include <cmath>
 
+CImage::~CImage() {
+    imageTimer->cancel();
+    imageTimer.reset();
+}
+
+static void onTimer(std::shared_ptr<CTimer> self, void* data) {
+    const auto PIMAGE = (CImage*)data;
+
+    PIMAGE->onTimerUpdate();
+    PIMAGE->plantTimer();
+}
+
+static void onAssetCallback(void* data) {
+    const auto PIMAGE = (CImage*)data;
+    PIMAGE->renderSuper();
+}
+
+std::string CImage::getUniqueResourceId() {
+    return std::string{"image:"} + std::to_string((uintptr_t)this) + ",path:" + path + ",time:" + std::to_string(std::chrono::system_clock::now().time_since_epoch().count());
+}
+
+void CImage::onTimerUpdate() {
+
+    if (!pendingResourceID.empty())
+        return;
+
+    if (!reloadCommand.empty()) {
+        path = g_pHyprlock->spawnSync(reloadCommand);
+
+        if (path.ends_with('\n'))
+            path.pop_back();
+
+        if (path.empty())
+            return;
+    }
+
+    // request new
+    request.id        = getUniqueResourceId();
```

Comment:
maybe fuckin stat the file first and check if it has actually changed instead of reloading it all the time??

Reply:
yeah a bit braindead rushed, I think just filesize is enough? what's the probability of same name + same size but different content

also don't think there is need to unload them at all, might be reused?

Reply:
just stat the modification date bro

> also don't think there is need to unload them at all, might be reused?

unload old when loading new.